pages/_app.js

```diff
@@ -1,12 +1,30 @@
 import GlobalStyle from "../styles";
 import { data } from "@/lib/db/db";
 import { useState } from "react";
-
+import { useRouter } from "next/router";
 export default function App({ Component, pageProps }) {
+  const router = useRouter();
+
   const [activities, setActivities] = useState(data);
   function addActivity(newActivity) {
     setActivities([{ ...newActivity, id: crypto.randomUUID() }, ...activities]);
   }
+
+  function removeActivity(id) {
+    setActivities(activities.filter((activity) => activity.id !== Number(id)));
```

Comment:
it would be easier if the id's in your `db.js` were also strings. 

I recommend changing the id in your `db.js` to strings and changing your removeActivity() as well as editActivity() to only compare with the id and not convert it to a type number beforehand.